### PR TITLE
Update word-count to streams bootstrap 3

### PR DESCRIPTION
**Pull request by raminqaf**

### CODE REVIEW

File: atm-fraud/defaults.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-kubernetes-app:
+helm-app:
```
Comment (disrupted) [inline atm-fraud/defaults.yaml:1]: unrelated?

File: word-count/pipeline.yaml [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
   values:
     image: bakdata/kpops-demo-sentence-producer
+    imageTag: "2.0.0"
 
-- type: streams-app-v2
-  name: word-counter
+- type: streams-app
+  name: word-count-app
+  values:
+    image: bakdata/kpops-demo-word-count-app
```
Comment (disrupted) [inline word-count/pipeline.yaml:10]: this depends on #19 to have it in a public Docker registry, right?

File: atm-fraud/defaults.yaml (hunk 1 above)
Comment (raminqaf) [reply atm-fraud/defaults.yaml:1]: I prefer using the `HelmApp` instead of the abstract `KubernetesApp` class

File: word-count/pipeline.yaml (hunk 2 above)
Comment (raminqaf) [reply word-count/pipeline.yaml:10]: That is true. I haven't changed anything in the `pipeline.yaml` other than the migration to streams-bootstrap v3. We should upload the new images manually.

Comment (disrupted) [reply word-count/pipeline.yaml:10]: I wonder how much effort it would be to just finish the CI PR. it would be really convenient to automatically upload those images.

File: atm-fraud/defaults.yaml (hunk 1 above)
Comment (philipp94831) [reply atm-fraud/defaults.yaml:1]: Why though? namespace is a Kubernetes related concept

File: word-count/code/src/test/java/com/bakdata/kpops/examples/SentenceProducerIntegrationTest.java [hunk 3]
```diff
@@ -59,11 +62,16 @@ void shouldRunApp() throws InterruptedException {
                 });
     }
 
-    private SentenceProducer setupApp() {
-        final SentenceProducer producerApp = new SentenceProducer();
-        producerApp.setBrokers(this.kafkaCluster.getBrokerList());
+    private KafkaProducerApplication<SentenceProducer> setupApp() {
+        final KafkaProducerApplication<SentenceProducer> producerApp = new KafkaProducerApplication<>() {
```
Comment (philipp94831) [inline word-count/code/src/test/java/com/bakdata/kpops/examples/SentenceProducerIntegrationTest.java:66]: You could reuse the SimpleProducerApplication you also use in main

File: word-count/defaults.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    kafka:
+      bootstrapServers: ${config.kafka_brokers}
+      schemaRegistryUrl: ${config.schema_registry.url}
+      optimizeLeaveGroupBehavior: false
+    prometheus:
+      jmx:
+        enabled: false
+
+    replicaCount: 1
+  debug: true
```
Comment (philipp94831) [inline word-count/defaults.yaml:18]: This doesn't exist anymore. Alsow it was nested under values

File: word-count/defaults.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+streams-bootstrap:
+  version: "3.1.0"
+  values:
+    labels:
+      pipeline: ${pipeline.name}
+    kafka:
+      bootstrapServers: ${config.kafka_brokers}
+      schemaRegistryUrl: ${config.schema_registry.url}
+      optimizeLeaveGroupBehavior: false
```
Comment (philipp94831) [inline word-count/defaults.yaml:12]: Why bother configuring it?

**Review by philipp94831 — CHANGES_REQUESTED** [review]

File: word-count/defaults.yaml (hunk 5 above)
Comment (raminqaf) [reply word-count/defaults.yaml:11]: We don't need SR 😄 The app only has (String, String) records.

Comment (philipp94831) [reply word-count/defaults.yaml:11]: But then remove optimizeLeaveGroupBehavior

Comment (raminqaf) [reply word-count/defaults.yaml]: Done
